### PR TITLE
fix(typeahead): add z-index to typeahead dropdown and update dropdown

### LINKED ISSUE #1631: Typeahead: menu layer is placed behind the modal overlay making it unusable
when a typeahead is used inside of the Dialog(modal=true) the typeahead menu falls behind the dialog in stacking order.

https://codesandbox.io/s/codesandboxer-example-forked-wqppx?file=/example.tsx

### PR DESCRIPTION
**Pull request by EdwardIrby**
resolves #1631

### CODE REVIEW

File: packages/typeahead/src/css/index.ts [hunk 1]
```diff
@@ -21,7 +22,8 @@ export default {
     maxWidth: '320px',
     minWidth: '160px',
     overflowY: 'auto',
-    padding: `${layout.spacingXXSmall} 0`
+    padding: `${layout.spacingXXSmall} 0`,
+    zIndex: layers.globalDropdown
```
Comment (jaketrent) [inline packages/typeahead/src/css/index.ts:26]: We probably have a lack of clarity on whether we should use these layers for global nav only or for other stuff in the DS.  Put another way, should we put in-page dropdowns at a lower z-index than the global nav, which is layered on top of the page app (in whole)?

Comment (EdwardIrby) [reply packages/typeahead/src/css/index.ts:26]: Oh didn't realize that. Hmm Probably @sweeting got a value?

Comment (EdwardIrby) [reply packages/typeahead/src/css/index.ts:26]: I think this is okay as is. Do you imagine a situation where this dropdown and another dropdown type component or the global nav dropdown will all be open and visible in the same space?

Comment (jaketrent) [reply packages/typeahead/src/css/index.ts:26]: Ok, that's fine. I don't have a better answer. I suppose when we run into the concrete problem, we can adjust.

Yes, I can see a scenario.  Top-left global nav profile menu is open while a typeahead table filter in the top-left of the page are open at the same time. `#ps-main` (the app) comes *after* the profile menu in `#ps-chrome`, so the app typeahead appears on top of the other items on the same z-index. Should it?  (assuming that seems plausible.) Definitely corner case and probably doesn't matter.

**Pull request merged**